api/src/main/java/com/capitalone/dashboard/service/ServiceServiceImpl.java

```diff
@@ -90,4 +101,49 @@ public void deleteDependentService(ObjectId dashboardId, ObjectId serviceId) {
         service.getDependedBy().remove(dashboardId);
         serviceRepository.save(service);
     }
+
+    @Override
+    public void refreshService(ObjectId dashboardId, ObjectId serviceId) {
+        Service service = get(serviceId);
+        LOGGER.info("URL is :" + service.getUrl());
+        ServiceStatus status = getServiceStatusBasedOnHttpReturnCode(getHttpCode(service.getUrl()));
+        service.setDashboardId(dashboardId);
+        service.setStatus(status);
+        service.setLastUpdated(System.currentTimeMillis());
+        Dashboard dashboard = dashboardRepository.findOne(dashboardId);
+        service.setApplicationName(dashboard.getApplication().getName());
+        serviceRepository.save(service);
+    }
+
+    private ServiceStatus getServiceStatusBasedOnHttpReturnCode(int code) {
+        ServiceStatus status = null;
+        if (code == 200) {
+            status = ServiceStatus.Ok;
+        } else if (code >= 300 && code <= 400) {
```

Comment:
@MarkRx open for discussion.. i have noticed with application having o-auth enabled you will get 403, so that should also be good. 401's also mean the application was reachable but the call was challenged. 
Looking for a more robust way - and maybe Hygieia community can help. 
The goal of this exercise was to make it useful in place of a dummy service widget.